### ISSUE TITLE
Split Table methods (and tests) out into their own files

### ISSUE THREAD

**Issue by onyxfish** [wireservice/agate#482]
The size of the main files is getting unmaintainable. Core Table ops should be split out into functional methods:

def distinct(table, columns)

Which are then invoked by the method, as the preview methods currently do.

Need to think about the best way to organize this and whether or not these functions should be exposed as part of the API.

**Comment by onyxfish** [comment]
This is reaching crisis proportions. `table.py` is almost 2000 lines long. Won't fix for the next release, but it needs to be done before the following one. Bumping to normal and assigning to myself.

**Comment by nbedi** [comment]
You beat me to posting the same comment. Core files are definitely too huge/intimidating at this point. Following release sounds good.

**Comment by onyxfish** [comment]
When the time comes it probably makes sense for us to create a branch and finer grained tickets so we can both work on this at the same time. I think it we'll want to follow the "preview" pattern where each table method becomes a function accepting `table` as the first argument. Currently leaning toward not exposing the additional functions in the API. ("one right way", etc.)

I think the painful part will be breaking up the tests. Fortunately, you've made a good start for us with the `TestPivot` style classes you've been creating.

**Comment by nbedi** [comment]
Sounds good. This will be a fun beast to tackle. "Preview" pattern looks good to me.

**Comment by onyxfish** [comment]
Can refactoring table methods be combined with streamlining the extension interface? Maybe every core set of table methods are an "extension" that is patched in. Except the patch mechanism is clumsy for this. Is there some sort of registration system that could work? Is this going to make the documentation a total mess?

**Comment by onyxfish** [comment]
Started this work on the flight yesterday. Ending up merging it directly to master to minimize impact on other bug fixing activities.

**Comment by onyxfish** [comment]
Major methods on both `Table` and `TableSet` are now composed from separate modules. On the fence on whether or not it's worth breaking out the simpler methods as well, but will probably end up doing it just for consistency. Tests still need to be refactored.

**Comment by nbedi** [comment]
Did some test refactoring to match your method refactoring. Mostly moving code around, may be room for further clean up.

**Comment by onyxfish** [comment]
Calling this done for the time being.
